src/main/java/frc/robot/subsystems/hang/docs/Hang.md

```diff
@@ -0,0 +1,68 @@
+|getHangVelocity|double|N/A|finds the velocity of the motor| |
+|getMotorCurrent|double|N/A|find the currunent being outputed in the motor|in amps|
+|setHangMode|void|HangControlType type|figures out the hang mode to determine how the motor will move|uses a enum class. the modes either run at a constant voltage (MANUAL mode) to test how the arm runs structualy, or the motor goes to a postion using PID (PID mode) for actual use within competition |
+
+## what is logged to dashboard
+| name | Description | Notes |
+| --- | --- | ---- |
+|kMotorInverted|logs if the motor is inverted|under setupMotors method|
+|kHangMotorLimit|logs the positon limit of the motor|under setupMotors method|
+|kPositionConversionFactor|logs the conversion factor of the motor position for the encoders|under setupMotors method|
+|kVelocityConversionFactor|logs the conversion factor of the motor velocity for the encoders|under setupMotors method|
+|kBrake|logs if the motors are braked(wont move) for idle mode|under setupMotors method|
+|Kp, Ki, Kd| logs the p,i,and d for motor soothness|under setupMotors method|
+| --- | --- | ---- |
+|Hang Current Velocity|logs the velocity of the motor|under doSendables method|
+|Hang Current Position|logs the position of the motor|under doSendables method|
+|Hang Target Position|logs the postion that is wanted for the motor to go to|under doSendables method|
+|Hang Error|logs the error of the motor (distance from wanted position)|under doSendables method|
+|Hang Current|logs the current of the motor|under doSendables method|
+|Hang Bus Voltage|logs the voltage of the canbus|under doSendables method|
+|Hang Output|logs the power output of the spark max|under doSendables method|
+| --- | --- | ---- |
+|kHangTolerance|logs the tolerance (the allowed error) of the motor|under isAtPosition method|
+
+# how to use
+to use the subsytem, you need to create a mountable motor connected to a sparkmax/encoder
+you would also need to have the frc dashboard to be able to tweak and modify code for desired function.
+
+## methods needed for the two modes
+### for Manual mode you would need to have 
+ - getInstance(to actualy run the motor), 
+ - setupMotors(to create/log initial values for testing), 
+ - setManualVelocity(to actualy move the motor), 
+ - getHangPosition(so the motor knows where it is), 
+ - setTargetPosition(to tell where we want the motor to go), 
+ - isAtPosition(to see if we can stop the motor from running), 
+ - getError(so the encoder knows how far it needs to go to stop), 
+ - getHangVelocity(to see if the motor is running correctly), 
+ - getMotorCurrent(to see if the motor is running correcly) and 
+ - setHangMode(to set the mode to manual).
+
+### -----------------------------------------------------
+
+### for PID mode you would need to have 
+- all the methods listed for the manual mode. in addition you would need 
+- setPID(to actually be able to smooth out movement of the motor) 
+
```

Comment:
look at above feedback. A lot of the methods you listed in manual are in fact methods for PID usage. What I said earlier about writing how they interact with each other comes into play here especially